File: src/main/java/net/gtaun/shoebill/amx/types/ReturnType.java
package net.gtaun.shoebill.amx.types;

/**
 * Created by marvin on 15.01.16.
 * Copyright (c) 2015 Marvin Haschker. All rights reserved.
 */
public enum ReturnType {
    INTEGER(0),
    FLOAT(1),
    STRING(2);

    private final int value;
    private ReturnType(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }
}

File: src/main/java/net/gtaun/shoebill/amx/types/ReturnType.kt
package net.gtaun.shoebill.amx.types

/**
 * This enums defines the possible return types that can be used by the Amx-Interface.
 * Since the abstract machine (AMX) does only support a few types, this enum is tiny.
 *
 * @author Marvin Haschker
 */
enum class ReturnType private constructor(val value: Int) {
    INTEGER(0),
    FLOAT(1),
    STRING(2)
}
